### PR TITLE
feat: Introducing Azure VNET encryption to module and examples

### PR DESCRIPTION
**Pull request by acelebanski**
## Description

This PR introduces [Azure Virtual Network encryption](https://learn.microsoft.com/en-us/azure/virtual-network/virtual-network-encryption-overview#limitations). Enabled by default on the module level, but disabled by default in examples.

By occasion, this PR also replaces deprecated properties in `vmseries` module:
- `enable_accelerated_networking` -> `accelerated_networking_enabled`
- `enable_ip_forwarding` -> `ip_forwarding_enabled`

## Motivation and Context

#52 

## How Has This Been Tested?

Locally from my workstation by deploying chosen examples in the lab.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes if appropriate.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by acelebanski** [comment]
/idempotence paths="examples/common_vmseries examples/common_vmseries_and_autoscale examples/dedicated_vmseries examples/dedicated_vmseries_and_autoscale examples/gwlb_with_vmseries examples/standalone_panorama examples/standalone_vmseries" tf_version="1.5 1.6 1.7"

**Comment by acelebanski** [comment]
/idempotence paths="examples/common_vmseries examples/common_vmseries_and_autoscale examples/dedicated_vmseries examples/dedicated_vmseries_and_autoscale examples/gwlb_with_vmseries examples/standalone_panorama examples/standalone_vmseries" tf_version="1.5"

**Review by alperenkose — COMMENTED** [review]
It looks good to me, I think we shall just make up some more clear naming for the variable.

File: modules/vnet/variables.tf [hunk 1]
```diff
@@ -49,6 +49,13 @@ variable "address_space" {
   }
 }
 
+variable "enable_vnet_encryption" {
```
[the remark this answers is not in the document]
Comment (acelebanski) [reply modules/vnet/variables.tf:52]: It's not what it does. This variable literally enables VNET Encryption. This feature forces encrypted communication between supported VMs & VMSSs inside the VNET. The "Allow Unencrypted" means that if VMs doesn't support Encryption, the traffic is not dropped (it's the only supported mode so far).

[the remark this answers is not in the document]
Comment (acelebanski) [reply modules/vnet/variables.tf]: Ok, I've redone it. The variable is now called `vnet_encryption` and is of string type.

[the remark this answers is not in the document]
Comment (sebastianczech) [reply modules/vnet/variables.tf]: I like it much better, thank you 👍

**Comment by acelebanski** [comment]
/idempotence paths="examples/common_vmseries examples/common_vmseries_and_autoscale examples/dedicated_vmseries examples/dedicated_vmseries_and_autoscale examples/gwlb_with_vmseries examples/standalone_panorama examples/standalone_vmseries"

**Review by sebastianczech — APPROVED** [review]
Great work 👍

**Pull request merged**